src/wms/src/main/java/org/geoserver/wms/svg/StreamingSVGMap.java

```diff
@@ -138,10 +140,10 @@ private void writeLayers() throws IOException {
 
                 String styleName = layer.getStyle().getName();
 
-                writer.write("<g id=\"" + groupId + "\"");
+                writer.write("<g id=\"" + escapeXml10(groupId) + "\"");
 
                 if (!styleName.startsWith("#")) {
-                    writer.write(" class=\"" + styleName + "\"");
+                    writer.write(" class=\"" + escapeXml10(styleName) + "\"");
```

Comment:
SVGWriter.encodeAttribute doesn't exist.  There is a AttributesSVGHandler.encodeAttribute but that is a private non-static method on a non-static inner class of SVGWriter and, even though it would prevent injection, it actually doesn't guarantee valid XML output.
SVGWriter has a lot dead code and I only updated code that could actually be run in a live GeoServer.

Reply:
These are examples of where a lot of the SVGWriter functionality gets commented out:
https://github.com/geoserver/geoserver/blob/main/src/wms/src/main/java/org/geoserver/wms/svg/SVGWriter.java#L142
https://github.com/geoserver/geoserver/blob/main/src/wms/src/main/java/org/geoserver/wms/svg/SVGWriter.java#L208
https://github.com/geoserver/geoserver/blob/main/src/wms/src/main/java/org/geoserver/wms/svg/SVGWriter.java#L250